src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/tools/HeapSummary.java

```diff
@@ -86,11 +86,7 @@ public void run() {
       printValMB("MetaspaceSize            = ", getFlagValue("MetaspaceSize", flagMap));
       printValMB("CompressedClassSpaceSize = ", getFlagValue("CompressedClassSpaceSize", flagMap));
       printValMB("MaxMetaspaceSize         = ", getFlagValue("MaxMetaspaceSize", flagMap));
-      if (heap instanceof ShenandoahHeap) {
-         printValMB("ShenandoahRegionSize     = ", ShenandoahHeapRegion.regionSizeBytes());
-      } else {
-         printValMB("G1HeapRegionSize         = ", HeapRegion.grainBytes());
-      }
+      printValMB("G1HeapRegionSize         = ", HeapRegion.grainBytes());
```

Comment:
This call to grainBytes() is suspect because it assumes G1 is being used even though it may not be. That was the case even before your changes because it was being called for SerialGC and ParallelGC. Perhaps the old code is better, but also fix it to only print G1HeapRegionSize for G1. I'd also suggest fixing the test to just remove the check for G1HeapRegionSize. It's not important.

Reply:
I am happy to just drop the check to `G1HeapRegionSize` in the test.